index.js

```diff
@@ -62,7 +62,12 @@ function templateCacheFiles(root, base, templateBody, transformUrl, escapeOption
     if (typeof base === 'function') {
       url = path.join(root, base(file));
     } else {
-      url = path.join(root, file.path.replace(base || file.base, ''));
+      if (base === undefined) {
+        // Make sure file.base ends in a separator, to ensure correct relative
+        // path construction (provides compatibility with Gulp 4/Vinyl 2)
+        base = file.base += file.base.substr(-1) === path.sep ? "" : path.sep;
```

Comment:
@simonua @dmellstrom this is a breaking change on my Gulp 3.9 configuration. My root is undefined and the `file.base` is an absolute folder path ending with backslash, so the line _68_ sets `base = ""` and then the line _70_ does not transform the absolute path `file.path` to a relative path.

Reply:
Can confirm this is an issue (gulp v3.9.1).

Reply:
@vanni-giachin & @davyjoe, bummer! My apologies for the breaking change. Given the still relatively high number of downloads of this package, I reverted and issued 2.2.5 to reestablish status quo prior to the breaking change.
@dmellstrom & @michaelcyyuen, I'm going to open the issues we closed with 2.2.4 again, so they can be tackled. I think the option for you is to lock into 2.2.4 (less desirable overall, really) or give it another go if you don't mind.

Reply:
@vanni-giachin @davyjoe I take it you are on Windows? (You mentioned a backslash.) Could you please give me more specific information about your configuration and the paths that are not working?
I haven't been able to replicate the regression you are describing on Windows/Gulp 3.9.1.
I can't see a possible value for `file.base` that would result in `base = ""`.

Reply:
@dmellstrom I am running on a Mac. Here is a sample of how my gulp task is structured:
```
'use strict';

var templateCache = require('gulp-angular-templatecache');

var TEMPLATE_HEADER = '\'use strict\'; angular.module(\'<%= module %>\'<%= standalone %>).run([\'$templateCache\', function($templateCache) {';
var TEMPLATE_PATHS = [
      'app/content/_doc1.html',
      'app/components/myComponent1/*.html',
      'app/components/myModal/_detail.html'
];
var TEMPLATE_MODULE = 'myModule';
var TEMPLATE_DESTINATION = 'app/scripts/templates';

gulp.task('templatecache', function () {
  return gulp.src(TEMPLATE_PATHS)
    .pipe(templateCache({ module: TEMPLATE_MODULE, templateHeader: TEMPLATE_HEADER }))
    .pipe(gulp.dest(TEMPLATE_DESTINATION));
});
```